File: src/BKE.Dna.Logger.Platform.Android/AndroidDnaPaths.cs
using Android.Content;

namespace BKE.Dna.Logger.Platform.Android;

internal static class AndroidDnaPaths
{
    public static string CaptureRoot(Context context)
    {
        ArgumentNullException.ThrowIfNull(context);
        var filesDirectory = context.FilesDir?.AbsolutePath
            ?? throw new InvalidOperationException("Android app files directory is unavailable.");

        return Path.Combine(filesDirectory, "dna", "captures");
    }
}

File: android/app/src/main/kotlin/com/bke/dna/logger/AndroidDnaPaths.kt
package com.bke.dna.logger

import android.content.Context
import java.io.File

object AndroidDnaPaths {
    fun capturesRoot(context: Context): File =
        File(context.filesDir, "dna/captures").also { root ->
            check(root.exists() || root.mkdirs()) {
                "Unable to create app-private DNA capture root"
            }
        }

    fun workingDataRoot(context: Context): File =
        File(context.filesDir, "dna/working-data").also { root ->
            check(root.exists() || root.mkdirs()) {
                "Unable to create app-private Working Data root"
            }
        }
}
